### DISCUSSION TITLE
Allow passing multiple files to commands

### DISCUSSION THREAD

**Discussion by mattyclarkson** [aquasecurity/trivy#4203] [Ideas]
We currently allow two types of invocation to commands:

`trivy <cmd> <single-dir> --files-patterns <pattern>`
`trivy <cmd> <single-file>`

It would be incredibly useful to specify multiple files:

`trivy <cmd> <single-file> <single-file>`

This would be especially useful to hook up Trivy into the [pre-commit](https://pre-commit.com/) framework as `pre-commit` passes multiple files to be checked to the tool and does not, and will not, [support passing single files](https://github.com/pre-commit/pre-commit/issues/1728#issuecomment-741901582).

With this implemented we can have `pre-commit` hooks for `trivy fs`/`trivy config`, etc to prevent violations ever getting into a repository.

**Comment by knqyf263** [comment]
I remember there was the same request. Let's see how much demand there is.
https://github.com/aquasecurity/trivy/issues/1162

**Reply by bober2000** [reply]
It seems to be closed :(

**Comment by bober2000** [comment]
We also need this functionality.
While using GitHub Action I want to check only changed files/directories as full scan is not needed and taking too much time

**Comment by wagde-orca** [comment]
i would also like to see this option to send a list of files (or a file that contains file paths) to scan.
another alternative is to support links, if trivy supports links we can generate a directory which contains links to the real files that should be scanned...
for example if we want to scan
/a/b/c/a.jar
/d/d.jar
we can create a directory called links which contains links to the files
links/a/b/c/a.jat -> /a/b/c/a.jar
links/d/d.jar -> /d/d.jar
and we can invoke trivy rootfs /links
trivy will resolve the links of the files and scan them

**Comment by vitalii-morvaniuk-airslate** [comment]
Any news on this?

**Comment by silverwind** [comment]
This would be very useful to target only specific files in the `fs` module:

```
trivy fs --scanners vuln uv.lock package-lock.json
```

Currently scanning multiple specific files requires multiple invocations of the command. I can't pass a directory in this case because I want to exclude a certain file and don't want to mess with `--files-patterns` which seems to hard to configure. Passing a explicit list of files would be ideal.
